### ISSUE TITLE
npm create astro@latest directory creation captures Ctrl-W "cut word" shortcut as control character

### ISSUE THREAD

**Issue by JohnMaguire** [withastro/astro#6609]
### What version of `astro` are you using?

2.1.3

### Are you using an SSR adapter? If so, which one?

N/A

### What package manager are you using?

npm

### What operating system are you using?

Mac

### What browser are you using?

N/A

### Describe the Bug

1. `npm create astro@latest` (`v2.1.3 Launch sequence initiated.`)
2. At prompt for project directory, press Ctrl-W, Ctrl-W to delete the contents.
3. Type ./blog followed by return key
4. Hit return through the rest of the prompts

```
❯ ls -l
total 8
drwxr-xr-x     3 jmaguire  staff      96 Mar 20 23:51 ???.
```

`cd [tab]` gives this suggestion: `❯ cd $'\027'$'\027'$'\027'.`

The directory contains the C-w control character.

### Link to Minimal Reproducible Example

n/a

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by ematipico** [comment]
@JohnMaguire, what's the expected behaviour from your point of view?

**Comment by JohnMaguire** [comment]
A directory called "blog" should be created at this level. Control codes should not make it into the path.

Ctrl-A and Ctrl-E do not seem to work (beginning / end of line) but do not create control codes in the path.

**Comment by andremralves** [comment]
I tested Vite's and Next's CLIs to see how they prevent this issue. Vite's CLI has the same behavior (creates this weird file) and Next's displays an info message saying that the name is in a invalid pattern.

So maybe a good idea would be to check for these invalid characters and display a message. Then the user can see that something went wrong and change the project name.

**Comment by JohnMaguire** [comment]
@andremralves That seems like much better behavior, though the ideal would be to properly handle the shortcuts. Ctrl-W actually has the expected effect, it just seems to _also_ insert control characters unexpectedly.

FWIW, these are "[emacs shortcuts](https://www.smartfile.com/blog/bash-shortcuts-for-the-command-line-emacs/)" which have been available in bash since... well, as long as I've used it (so pre-2005.) I don't know if they are implemented by bash/zsh/etc. or GNU readline or what. I just know that every interactive shell command I can think of off the top of my head right now supports them (I just tested zsh, fzf, weechat, vim, and cat.)

It does appear to me that there is [an attempt to detect them](https://github.com/withastro/cli-kit/blob/main/src/prompt/util/action.ts#L4-L17) and if I understand the code correctly, which I am only skimming, [a way to register actions to these commands.](https://github.com/withastro/cli-kit/blob/main/src/prompt/prompt.js#L10-L13) A cursory read also suggests that actions w/o callbacks will be ignored. I think this is why the other emacs shortcuts I tried do not work.

If that is the case, a simple solution is to register the C-w shortcut, which will break its functionality but at least avoid inserting erroneous control codes.

**Comment by JohnMaguire** [comment]
It looks like the bug is shared between vite and astro because vite uses this project, which cli-kit seems to be heavily based off of: https://github.com/terkelg/prompts/blob/master/lib/util/action.js

**Comment by JohnMaguire** [comment]
I filed https://github.com/terkelg/prompts/issues/394 over there, and https://github.com/terkelg/prompts/pull/330/files looks like it would indeed solve this issue by disabling the Ctrl+W shortcut entirely.

**Comment by andremralves** [comment]
@JohnMaguire Yes, I was looking into the code base and saw that the problem comes from the cli-kit too. I think that in the create-astro package the best that can be done is add a validation

I will see if I can do something in the cli-kit as well, but for now I think this solution that I came up with will be good for preventing the user from creating files with non-printable chars. I filed a PR, let's see if they merge it :)

**Comment by JohnMaguire** [comment]
@andremralves Thanks for filing a PR in cli-kit! I must ask though - is there a reason you didn't include this line from the original PR made to terkelg/prompts? https://github.com/terkelg/prompts/pull/330/files#diff-384dd446a9beb22e86702df030f20bf8269923635e47ab56350b2de765baaeb9R40

This would disable the C-w shortcut entirely, avoiding an annoying iteration of "Your file name was invalid, try again!" :)

**Comment by andremralves** [comment]
@JohnMaguire, I didn't filed a PR for the cli-kit, I filed a PR in this repo for the `create-astro` package just to to add a validation checking for the presence of non-printable characters in the input. I think this can be a general solution that solves this problem even for when someone copy and paste a name with these invalid characters without typing Ctrl.

But I think your solution of disabling the Ctrl characters is a good one as well, I didn't filed it in the cli-kit bacause I thought you would like to file it yourself as you had the idea.

>  avoiding an annoying iteration of "Your file name was invalid, try again!"
I think it wouldn't be annoying because is something rare to happen.

**Comment by JohnMaguire** [comment]
@andremralves I'm a little confused. Is this your PR? https://github.com/withastro/cli-kit/pull/15

That PR is filed with cli-kit, and looks to mirror https://github.com/terkelg/prompts/pull/330/files#diff-384dd446a9beb22e86702df030f20bf8269923635e47ab56350b2de765baaeb9R40 albeit without the part that disables C-w.

>> avoiding an annoying iteration of "Your file name was invalid, try again!"
>
> I think it wouldn't be annoying because is something rare to happen.

Well, this might be a matter of opinion but I think it probably happens often. The prompt is pre-filled with text, so the natural inclination for someone familiar with these shortcuts (which have been around for probably decades) is to delete the text, the quickest method of which is to use C-w. It's almost reflexive.
